compiler/describe.go

```diff
@@ -0,0 +1,30 @@
+package compiler
+
+import (
+	"context"
+	"errors"
+
+	"github.com/brimdata/zed/compiler/data"
+	"github.com/brimdata/zed/compiler/describe"
+	"github.com/brimdata/zed/compiler/parser"
+	"github.com/brimdata/zed/compiler/semantic"
+	"github.com/brimdata/zed/lakeparse"
+)
+
+func Describe(ctx context.Context, query string, src *data.Source, head *lakeparse.Commitish) (*describe.Info, error) {
+	seq, sset, err := Parse(query)
+	if err != nil {
+		return nil, err
+	}
+	if len(seq) == 0 {
+		return nil, errors.New("internal error: AST seq cannot be empty")
+	}
```

Comment:
Nit: In line with the policy described at https://github.com/brimdata/zed/pull/5126#discussion_r1612923925, drop this?
```suggestion
```